File: foliage-api/src/main/java/org/lavenderx/foliage/api/ApiBootstrap.java
package org.lavenderx.foliage.api;

import lombok.extern.slf4j.Slf4j;
import org.lavenderx.foliage.api.web.WebServer;

import javax.servlet.ServletException;
import java.io.IOException;

@Slf4j
public class ApiBootstrap {

    private static final String SERVER_PORT = "foliage.server.port";

    public static void main(String[] args) {
        try {
            int port;
            if (System.getProperty(SERVER_PORT) == null) {
                port = 8080;
            } else {
                port = Integer.parseInt(System.getProperty(SERVER_PORT));
            }

            WebServer webServer = new WebServer("FoliageServer", port).start();

            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
                try {
                    webServer.destroy();
                } catch (final Exception ex) {
                    log.error("An exception occurred when destroying server", ex);
                }
            }));
        } catch (IOException | ServletException ex) {
            log.error("", ex);
        }
    }
}

File: foliage-api/src/main/kotlin/org/lavenderx/foliage/api/ApiBootstrap.kt
package org.lavenderx.foliage.api

import org.lavenderx.foliage.api.logging.loggerFor
import org.lavenderx.foliage.api.web.WebServer

object ApiBootstrap {
    private val SERVER_PORT = "foliage.server.port"
    private val logger = loggerFor<ApiBootstrap>()

    fun start() {
        try {
            val port = if (System.getProperty(SERVER_PORT) == null) 8080 else System.getProperty(SERVER_PORT).toInt()
            val webServer = WebServer("FoliageServer", port).start()

            Runtime.getRuntime().addShutdownHook(Thread {
                try {
                    webServer.destroy()
                } catch (e: Exception) {
                    logger.error("An exception occurred when destroying server", e)
                }
            })
        } catch (e: Exception) {
            logger.error("", e)
        }
    }
}

fun main(args: Array<String>) {
    ApiBootstrap.start()
}